src/index.ts

```diff
@@ -854,6 +870,64 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return !code || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry) {
+			return await fnCall();
+		}
+
+		const defaultRetry = this.retry || {};
+		const minDelayMs = retry.minDelayMs ?? defaultRetry.minDelayMs ?? 0;
+		const maxDelayMs = retry.maxDelayMs ?? defaultRetry.maxDelayMs ?? 0;
+		const maxAttempts = retry.maxAttempts ?? defaultRetry.maxAttempts ?? 0;
+
+		if (
+			minDelayMs <= 0 ||
+			maxDelayMs <= 0 ||
+			maxAttempts <= 0 ||
+			minDelayMs > maxDelayMs
+		) {
+			console.warn('pinejs-client invalid retry parameters');
+			return await fnCall();
+		}
+
+		const onRetryHandler = retry.onRetry ?? defaultRetry.onRetry;
+
+		let attempt = 0;
+		let prevErr: any;
+
+		const canRetryHandler =
+			retry.canRetry ?? defaultRetry.canRetry ?? this.canRetryDefaultHandler;
+
+		while (attempt < maxAttempts) {
+			try {
+				if (attempt > 0) {
```

Comment:
So, I find it distasteful to put a conditional within a loop looking for one particular value... but it was a tradeoff in order to keep complexity down by calling this code from just one place and keeping the loop simple.  Another option would be to put this code into a local function and then separate the initial call from the retry loop.   Note that performance-wise, this was effectively what was happening in the previous version of this code but it was divided across two methods.  Feedback welcome.